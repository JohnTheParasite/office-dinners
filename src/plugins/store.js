import store from "@/store/store"

export const Store = {
  install(Vue) {
    Vue.prototype.$store = store
  }
}
