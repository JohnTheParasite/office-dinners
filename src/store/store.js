import Vue from "vue"
import Vuex from "vuex"
import global from "./modules/global"
import toasts from "./modules/toasts"

Vue.use(Vuex)

export default new Vuex.Store({
  strict: true,
  state: {},
  mutations: {},
  actions: {},
  modules: {
    global,
    toasts
  }
})
