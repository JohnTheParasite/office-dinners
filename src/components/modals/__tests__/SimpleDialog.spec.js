import { createLocalVue, shallowMount } from "@vue/test-utils"
import simpleDialog from "@/components/modals/SimpleDialog"
import i18n from "@/i18n"
import store from "@/store/store"
import { BootstrapVue } from "bootstrap-vue"

describe("modal for questions", () => {
  const localVue = new createLocalVue()
  localVue.use(i18n)
  localVue.use(store)
  localVue.use(BootstrapVue)

  it("should render SimpleDialog", () => {
    let dialog = shallowMount(simpleDialog, { localVue, i18n, store })
    expect(dialog).toMatchSnapshot()
  })

  it("test showDialog watcher", async () => {
    let dialog = shallowMount(simpleDialog, { localVue, i18n, store })
    expect(dialog.vm.showDialog).toBeFalsy()
    store.commit("dialog/openDialog")
    await dialog.vm.$nextTick()
    expect(dialog.vm.showDialog).toBeTruthy()
    await dialog.vm.$nextTick()
    store.commit("dialog/hideDialog")
    expect(dialog.vm.showDialog).toBeFalsy()
  })

  it("test hide and show methods", () => {
    let dialog = shallowMount(simpleDialog, { localVue, i18n, store })
    dialog.vm.show()
    store.commit("dialog/openDialog")
    expect(dialog.vm.showDialog).toBeTruthy()
    dialog.vm.hide()
    expect(dialog.vm.showDialog).toBeFalsy()
  })
})
