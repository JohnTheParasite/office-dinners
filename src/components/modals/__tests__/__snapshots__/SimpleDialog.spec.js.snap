// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal for questions should render SimpleDialog 1`] = `
<b-modal-stub id="b6" canceltitle="Cancel" cancelvariant="secondary" centered="true" headerclass="primary" headerclosecontent="&amp;times;" headercloselabel="Close" nocloseonbackdrop="true" oktitle="OK" okvariant="primary" size="md" titletag="h5">
  <div class="content">

  </div>
</b-modal-stub>
`;
